Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -0,0 +1,40 @@
+# Proposal for meta/run.yaml content being present in the composed image
+
+Currently, Capstan runtime support makes use of `$CWD/meta/run.yaml` for composing and running an OSv image. Although this is fine for the composing, it gets tricky for running because `capstan run <IMAGE>` generally does not require user to be inside package root directory.
+
+I propose the following solution:
+
+### Composing image
+File meta/run.yaml is present at package root directory (same as until now)
+
+When user runs `capstan package compose` the content of meta/run.yaml file is processed and new files are generated inside the OSv image folder `/osv-runtime`:
+
+Capstan should create a separate file for each named configuration e.g.
+```
+/osv-runtime
+   |--burek
+   |--pizza   <--------\
+   |                    |
+   |--default-runtime  /
+```
+Each runtime file should contain only one line representing the run command that is to be run inside OSv:
+```
+# content of file /osv-runtime/burek:
+node burek.js
+```
+
+When user runs `capstan package compose` the boot command is set so that the agent is run. A capstan package must be implemented that provides agent funcionality.
```

Comment:
The main part that confused me is "the boot command is set so that the agent is run". Currently, if I run `capstan run -e "cmd-to-execute"` this command will overwrite whatever the command was previously marked by Capstan. If I then upload such an image to OpenStack, it will actually run the `cmd-to-execute`. This can happen while the package maintainer would like to test the new package before publishing it.

Thus, please be aware of this that the commend line that starts the agent should not use the existing mechanism, otherwise people will quickly override it. Below I discussed about cloud-init's `runcmd` which would be executed regardless of the actual command.